content/ui-patterns/notification-messaging.mdx

```diff
@@ -0,0 +1,174 @@
+
+<img
+  width="960"
+  alt="An example of an input field with a 12px error message next to two examples of 14px icon usage, which uses an outline icon."
+  src="https://github.com/primer/design/assets/18661030/6ab6f9e9-d7e9-4f75-907f-0e08aec0db3c"
+/>
+
+### Overview
+
+Messaging components combine state with message type to provide the appropriate level of prominence. To determine which component to use for your message, start by identifying the message state.
+
+| State | Use case |
+|--------|--------|
+| <Box sx={{color: 'accent.fg', display: 'flex', alignItems: 'center', gap: '0.5rem'}}><InfoIcon /><span>Info</span></Box> | Highlights important information that has an influence on the current view or offers an action. |
+| <Box sx={{color: 'success.fg', display: 'flex', alignItems: 'center', gap: '0.5rem'}}><CheckCircleIcon /><span>Success</span></Box> | Informs about successfully completing an action. |
+| <Box sx={{color: 'attention.fg', display: 'flex', alignItems: 'center', gap: '0.5rem'}}><StopIcon /><span>Warning</span></Box> | Informs about a potential issue or consequence as the result of an action. |
+| <Box sx={{color: 'danger.fg', display: 'flex', alignItems: 'center', gap: '0.5rem'}}><AlertIcon /><span>Critical</span></Box> | Informs about an error that occured or warns about loss of access or data as a result of an action. |
+| <Box sx={{color: 'done.fg', display: 'flex', alignItems: 'center', gap: '0.5rem'}}><InfoIcon /><span>Upsell</span></Box> | Highlights features that are available for specific account types such as Enterprice. |
+
+### Info
+
+Use the `info` state to highlight messages that help users complete a task or that provide additional context.
+
+Examples:
+- Single sign on or other account related notifications
+- General information about a feature or setting
+
+<img
+  width="960"
+  alt="A single sign on info banner on the GitHub homepage"
+  src="https://github.com/primer/design/assets/18661030/1055049e-6db8-41d5-8cf8-e375bd0e8dea"
+/>
+
+### Warning
+
+Use the `warning` state to inform users of a potential issue, an important upcoming change that may impact their experience or a potentially unexpected consequence of an action.
+
+Examples:
+- An account setting will become required soon
+- A [degraded experience](./degraded-experiences) due to connectivity or GitHub availability
+- A non-destructive action with side-effects like costs or access changes is about to take place
+
+<img
+  width="960"
+  alt="A global warning banner above GitHub's global navigation bar inside a browser window."
+  src="https://github.com/primer/design/assets/18661030/468b73b1-de8b-45e0-b8bb-325516df942e"
+/>
+
+### Success
+
+Use the `success` state to notify users that the results of an action were successful if it is not apparent from other parts of the UI. **Use success messaging sparingly** and rely more on interaction context.
+
+Examples:
+- Saving an account setting if there's no page reload or redirect
```

Comment:
Does this means that:

- We should *not* render a success banner when saving a setting leads to a redirect / page reload? 
- Should we remove these instances from the product, where we consider the redirect/reload a clear indication of the success?